cosmetics-web/spec/services/notification_delete_service_spec.rb

```diff
@@ -0,0 +1,84 @@
+require "rails_helper"
+
+# rubocop:disable Style/RescueModifier
+RSpec.describe NotificationDeleteService do
+  let(:responsible_person) { create(:responsible_person) }
+  let(:submit_user) { create(:submit_user) }
+  let(:notification) { create(:registered_notification, cpnp_reference: "123412344") }
+  let(:current_user) { submit_user }
+  let(:other_user) { submit_user }
+  let(:service) { described_class.new(notification, current_user) }
+
+  before do
+    freeze_time
+    notification
+    create(:responsible_person_user, user: submit_user, responsible_person: responsible_person)
+  end
+
+  it "deletes the notification" do
+    expect {
+      service.call
+    }.to change(Notification, :count).by(-1)
+  end
+
+  it "creates log" do
+    expect {
+      service.call
+    }.to change(NotificationDeleteLog, :count).by(1)
+  end
+
+  it "creates log entry with correct data" do
+    service.call
+
+    log = NotificationDeleteLog.first
+
+    expect(log.submit_user).to eq submit_user
+    expect(log.notification_product_name).to eq notification.product_name
+    expect(log.responsible_person).to eq notification.responsible_person
+    expect(log.notification_created_at).to eq notification.created_at
+    expect(log.notification_updated_at).to eq notification.updated_at
+    expect(log.cpnp_reference).to eq notification.cpnp_reference
```

Comment:
RSpec provides a nice helper for avoiding the multiple expectations violation in cases like this:
```suggestion
    expect(log).to have_attributes(
      submit_user: submit_user,
      notification_product_name: notification.product_name,
      responsible_person: notification.responsible_person,
      notification_created_at: notification.created_at,
      notification_updated_at: notification.updated_at,
      cpnp_reference: notification.cpnp_reference,
    )
```